src/notebooks/controllers/vscodeNotebookController.ts

```diff
@@ -100,6 +101,7 @@ export class VSCodeNotebookController implements Disposable {
      */
     public static kernelAssociatedWithDocument?: boolean;
     private isDisposed = false;
+    private runningCellExecution: NotebookCellExecution | undefined;
```

Comment:
Is this mapping correct? I would think it's one runningCellExecution per NotebookDocument. The controller could have multiple cells running.

Reply:
Very good catch. I didn't even think of that.